src/features/authentication/login/login.tsx

```diff
@@ -18,138 +17,240 @@ import "./login.scss";
+          auth,
+          formValues.email,
+          formValues.password
+        );
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+        if (error.message.includes("user-not-found")) {
+          errors.email = "Email is incorrect.";
+          setFormErrors(errors);
+        } else if (error.message.includes("password")) {
+          errors.password = "Password is incorrect.";
+          setFormErrors(errors);
+        }
+      }
     };
 
-    const register = () => {
-      createUserWithEmailAndPassword(auth, email, password)
-        .then((userAuth) => {
-          console.log("User created");
-        })
-        .catch((err) => {
-          alert(err);
-        });
+    const signInWithGoogle = async () => {
+      const provider = new GoogleAuthProvider();
+      try {
+        await signInWithPopup(auth, provider);
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+      }
+    };
+
+    const signInWithFacebook = async () => {
+      const provider = new FacebookAuthProvider();
+      try {
+        await signInWithPopup(auth, provider);
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+      }
+    };
+
+    const logout = async () => {
+      await signOut(auth);
+      localStorage.setItem("authenticated", JSON.stringify(false));
+    };
+
+    const handleChange = (e: any) => {
+      const { name, value } = e.target;
+      setFormValues({ ...formValues, [name]: value });
+    };
+
+    const handleSubmit = (e: any) => {
+      e.preventDefault();
+      setFormErrors(validate(formValues));
+      setIsSubmit(true);
+    };
+
+    const validate = (values: any) => {
+      const errors = initialValues;
+
+      const regex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/i;
+      if (!values.email) {
+        errors.email = "Email is required.";
+      } else if (!regex.test(values.email)) {
+        errors.email = "This is not a valid email format!";
+      }
+      if (!values.password) {
+        errors.password = "Password is required.";
+      } else if (values.password.length < 8) {
+        errors.password = "Password must be more than 8 characters.";
+      }
+      return errors;
     };
 
     useEffect(() => {
-      onAuthStateChanged(auth, (userAuth) => {
-        if (userAuth) {
-          loginStore.loginSucces(userAuth);
-          console.log(userAuth);
-        } else {
-          loginStore.loginError();
-          console.log("ERROR ON LOG IN");
-        }
+      // validating form errors to see if we can loggin to app
+      if (!formErrors.email.length && !formErrors.password.length && isSubmit) {
+        loginToApp();
+      }
+    }, [formErrors]);
```

Comment:
I think that the state `isSubmit` should be added as a dependency as well in this effect to make sure it is fired when the user submits with no errors on the first try and also to cover other edge cases.